streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -584,6 +579,7 @@ public boolean test(final K1 key, final V1 value) {
         String name = topology.newName(LEFTJOIN_NAME);
 
         topology.addProcessor(name, new KStreamKTableLeftJoin<>((KTableImpl<K, ?, V1>) other, joiner), this.name);
+        topology.connectProcessorAndStateStores(name, other.getStoreName());
```

Comment:
If the input is materialized (like for sources), we use a `KTableSourceValueGetter` which does access the store. Or do I miss something?

I guess, a simple `null` check will do, right?